### PR TITLE
feat: support config sql metastore using username and password by env

### LINKED ISSUE #16998: feat: allow passing meta store SQL URL username and password by env
### Describe the bug

The meta store uses `sqlx` url string to connect the SQL backend. It acquires special characters to be encoded in HTML URL encoding.

### Error message/log

_No response_

### To Reproduce

_No response_

### Expected behavior

Support pass username and password via ENV and avoid extra encoding.

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolve #16998 .

1. support config `sqlite`, `postgres` and `mysql` in `--backend`.
2. add new configs correspondingly: `--sql-username`, `--sql-password` and `--sql-database`, which only required for  `postgres` and `mysql`.
3. `--sql-endpoint` should be configured in format `{host}:{port}` if the backend is set to  `postgres` and `mysql`. For `sqlite` it should be the file path.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

Support configure sql metastore using username, password and database separately. The corresponding envs are also supported.

1. support config `sqlite`, `postgres` and `mysql` in `--backend`.
2. add new configs correspondingly: `--sql-username`, `--sql-password` and `--sql-database`, which only required for  `postgres` and `mysql`.
3. `--sql-endpoint` should be configured in format `{host}:{port}` if the backend is set to  `postgres` and `mysql`. For `sqlite` it should be the file path.

### CODE REVIEW

File: src/risedevtool/src/task/meta_node_service.rs [hunk 1]
```diff
@@ -123,16 +123,18 @@ impl MetaNodeService {
                 is_persistent_meta_store = true;
 
                 cmd.arg("--backend")
-                    .arg("sql")
```
Comment (fuyufjh) [inline src/risedevtool/src/task/meta_node_service.rs:126]: Does the previous syntax still work? I am concerning about backward compatibility

Comment (yezizp2012) [reply src/risedevtool/src/task/meta_node_service.rs:126]: Yes, still works. The sql option is remained.

File: src/meta/src/backup_restore/restore.rs [hunk 2]
```diff
@@ -43,6 +43,15 @@ pub struct RestoreOpts {
     pub meta_store_type: MetaBackend,
     #[clap(long, default_value_t = String::from(""))]
     pub sql_endpoint: String,
+    /// Username of sql backend, required when meta backend set to MySQL or PostgreSQL.
+    #[clap(long, default_value = "")]
+    pub sql_username: String,
+    /// Password of sql backend, required when meta backend set to MySQL or PostgreSQL.
+    #[clap(long, default_value = "")]
+    pub sql_password: String,
```
Comment (kwannoel) [inline src/meta/src/backup_restore/restore.rs:51]:
```suggestion
    pub sql_password: Secret<String>,
```

Does this need to be secret as well?

Comment (yezizp2012) [reply src/meta/src/backup_restore/restore.rs:51]: I'm not sure. Do we need it? @zwang28

File: src/meta/node/src/lib.rs [hunk 3]
```diff
@@ -227,6 +239,36 @@ pub fn start(opts: MetaNodeOpts) -> Pin<Box<dyn Future<Output = ()> + Send>> {
                     .expose_secret()
                     .to_string(),
             },
+            MetaBackend::Sqlite => MetaStoreBackend::Sql {
+                endpoint: format!(
+                    "sqlite://{}?mode=rwc",
```
[the remark this answers is not in the document]
Comment (yezizp2012) [reply src/meta/node/src/lib.rs:244]: "rwc" is equivalent to setting both `SQLITE_OPEN_READWRITE` and `SQLITE_OPEN_CREATE`. We need them all for metastore, so it should be always required.

[the remark this answers is not in the document]
Comment (yezizp2012) [reply src/meta/node/src/lib.rs:244]: FYI, other modes include the following:
1. `ro` means read-only.
2. `rw` means read-write (but not create) access.
3. `memory` means using a pure in-memory database that never reads or writes from disk.

**Review by fuyufjh — APPROVED** [review]

File: src/meta/node/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (kwannoel) [reply src/meta/node/src/lib.rs:244]: Thought it was risingwave cloud :')

**Pull request merged**